src/frontend/src/handler/alter_source_column.rs

```diff
@@ -106,10 +106,7 @@ pub async fn handle_alter_source_column(
             catalog.definition =
                 alter_definition_add_column(&catalog.definition, column_def.clone())?;
             let mut bound_column = bind_sql_columns(&[column_def])?.remove(0);
-            bound_column.column_desc.column_id = columns
-                .iter()
-                .fold(ColumnId::new(i32::MIN), |a, b| a.max(b.column_id()))
-                .next();
+            bound_column.column_desc.column_id = max_column_id(columns).next();
```

Comment:
You can see `ALTER SOURCE ADD COLUMN` use this solution. Actually `bind_sql_columns ` previously takes `col_id_gen` as a param, but it's changed and use `ColumnId::placeholder()` after the refactor :( https://github.com/risingwavelabs/risingwave/pull/10307